src/git/GitExecutor.ts

```diff
@@ -188,4 +232,31 @@ export class GitExecutor {
             throw new Error(`Error executing git command: ${command}`);
         }
     }
+
+    /**
+     * Executes a raw Git command.
+     *
+     * @param {string} removePath - Path to remove.
+     */
+    public removePathFromSparseCheckout(removePath: string) {
+        const filePath = path.join(`${this.repoPath}/.git/info/sparse-checkout`);
+        fs.readFile(filePath, 'utf-8', (err, data) => {
+            if (err) {
+                console.error('Error at read file:', err);
+                return;
+            }
+
+            // Remove duplicates lines and the path
+            const lines = data.split('\n');
+            const uniqueLines = Array.from(new Set(lines)).filter((line) => {
+                return !line.includes(removePath);
```

Comment:
If the name of any file is substring of the name of another, both will be deleted.